src/eventstore/EventStoreSubscriber.cs

```diff
@@ -5,125 +5,72 @@
-        /// <param name="maxLiveQueueSize">Maximum size of the live queue.</param>
-        [Obsolete("Please use new static method CreateCatchUpSubscirptionFromPosition, this constructor will be removed in the future.")]
-        public EventStoreSubscriber(
-            IEventStoreConnection connection,
+        private EventStoreSubscriber(
+            EventStoreClient eventStoreClient,
             IDispatcher<ResolvedEvent> dispatcher,
             string streamName,
             ILogger logger,
-            int? startingPosition,
-            int catchUpPageSize = 1024,
-            int upperQueueBound = 2048,
-            TimeSpan? heartbeatFrequency = null,
-            TimeSpan? heartbeatTimeout = null,
-            int maxLiveQueueSize = 10000)
-            => Init(connection, dispatcher, streamName, logger, heartbeatFrequency, heartbeatTimeout, startingPosition, catchUpPageSize, upperQueueBound, maxLiveQueueSize);
+            ulong? startingPosition,
+            ulong liveEventThreshold)
+            => Init(eventStoreClient, dispatcher, streamName, logger, liveEventThreshold, startingPosition);
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="EventStoreSubscriber"/> class.
-        /// </summary>
-        /// <param name="connection">Eventstore connection.</param>
-        /// <param name="dispatcher">Dispatcher.</param>
-        /// <param name="streamName">Stream name to push events into.</param>
-        /// <param name="logger">Logger.</param>
-        /// <param name="checkpointFilePath">Path of the checkpoint file.</param>
-        /// <param name="catchupPageSize">Catchup page size.</param>
-        /// <param name="upperQueueBound">Upper Queue Bound.</param>
-        /// <param name="heartbeatFrequency">Frequency of heartbeat.</param>
-        /// <param name="heartbeatTimeout">Timeout of heartbeat.</param>
-        /// <param name="maxLiveQueueSize">Maximum size of the live queue.</param>
-        [Obsolete("Please use new static method CreateCatchupSubscriptionUsingCheckpoint, this constructor will be removed in the future.")]
-        public EventStoreSubscriber(
-            IEventStoreConnection connection,
+        private EventStoreSubscriber(
+            EventStoreClient eventStoreClient,
             IDispatcher<ResolvedEvent> dispatcher,
             ILogger logger,
             string streamName,
             string checkpointFilePath,
-            int catchupPageSize = 1024,
-            int upperQueueBound = 2048,
-            TimeSpan? heartbeatFrequency = null,
-            TimeSpan? heartbeatTimeout = null,
-            int maxLiveQueueSize = 10000)
+            ulong liveEventThreshold)
         {
-            _checkpoint = new WriteThroughFileCheckpoint(checkpointFilePath, "lastProcessedPosition", false, -1);
+            _checkpoint = new WriteThroughFileCheckpoint(checkpointFilePath, "lastProcessedPosition", false, StreamPosition.Start);
```

Comment:
Have dug into the docs and we need to use `FromStream.Start` (null) rather than `StreamPosition.Start` (0) here to make sure we get all the events. Subscription start position is still exclusive, so this will currently skip event 0.

Reply:
So WriteThroughFileCheckpoint would be modified to take the FromStream. Not sure how to implement this with the current logic. What do we write if it's null?

Reply:
@rcknight looks like we already use FromStream.Start in the actual subscription, it's just how we store it in our WriteThroughFileCheckpoint thing that we need to address here. it seems like ideally we'd write null (which is what FromStream.Start sets the stream position to) but I'm not sure if that's possible with the `WriteThroughFileCheckpoint` code.

Reply:
Does this still need to be set differently ?